src/reconcile/rec_write.c

```diff
@@ -1538,27 +1532,31 @@ __rec_upd_select(WT_SESSION_IMPL *session, WT_RECONCILE *r, WT_INSERT *ins,
 			if (upd->txnid != WT_TXN_NONE &&
 			    WT_TXNID_LT(upd->txnid, r->unstable_txn))
 				r->unstable_txn = upd->txnid;
+
 			/*
-			 * The durable timestamp is always set, and usually
-			 * the same as the start timestamp, which makes it OK
-			 * to use the two independently and be confident both
-			 * will be set.
+			 * The durable timestamp is always set by commit, and
+			 * usually the same as the start timestamp, which makes
+			 * it OK to use the two independently and be confident
+			 * both will be set.
 			 */
 			WT_ASSERT(session,
 			   upd->prepare_state == WT_PREPARE_INPROGRESS ||
 			   upd->durable_ts >= upd->start_ts);
 
-			if (upd->start_ts < r->unstable_timestamp)
+			if (r->unstable_timestamp > upd->start_ts)
 				r->unstable_timestamp = upd->start_ts;
+
 			/*
-			 * Don't set the unstable durable timestamp with the
-			 * durable timestamp of an in-progress prepared update.
 			 * An in-progress prepared update will always have a
-			 * zero durable timestamp.
+			 * zero durable timestamp.  Checkpoints can only skip
+			 * reading lookaside history if all updates are in the
+			 * future, including the prepare, so including the
+			 * prepare timestamp instead.
 			 */
-			if (upd->prepare_state != WT_PREPARE_INPROGRESS &&
```

Comment:
How did you find this one? It'd be nice to have a repeatable process for finding this sort of thing..

Reply:
I started down the path of including prepare timestamps in all of the "unstable" calculations, but then realized it isn't sufficient for `skew_newest` (because any later reader needs to get a prepare conflict).

I don't know any repeatable way to find these things other than increasing test coverage :-(